linode/resource_linode_object_storage_bucket.go

```diff
@@ -85,6 +188,31 @@ func resourceLinodeObjectStorageBucketRead(d *schema.ResourceData, meta interfac
 		return fmt.Errorf("failed to find the access config for the specified Linode ObjectStorageBucket: %s", err)
 	}
 
+	// Funtionality requiring direct S3 API access
+	if accessKey != "" && secretKey != "" {
```

Comment:
Instead of going off the keys alone here, we should see if lifecycleRule is present or versioning is enabled, then try to set these on the bucket.

Reply:
like ideally, we would thrown an error about not having the keys but having one of these fields present.